### PR TITLE
feat(webpack-cli): added mode argument

### PR DESCRIPTION
**Pull request by ematipico**
**What kind of change does this PR introduce?**

Feature, it adds back `mode` argument

**Did you add tests for your changes?**
Yes

**If relevant, did you update the documentation?**
Yes 

**Summary**
People asked to have back the `mode` argument.

**Does this PR introduce a breaking change?**

No

**Other information**

### CODE REVIEW

File: test/mode/prod/prod.test.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+'use strict';
+const { run } = require('../../utils/test-utils');
+const { stat } = require('fs');
+const { resolve } = require('path');
+describe('mode flags', () => {
+    it('should load a development config when --prod is passed', done => {
```
Comment (anshumanv) [inline test/mode/prod/prod.test.js]:
```suggestion
    it('should load a production config when --prod is passed', done => {
```

Comment (anshumanv) [reply test/mode/prod/prod.test.js]: Similarly in all tests below

File: test/mode/prod/prod.test.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        expect(stdout).toBeTruthy();
+        stat(resolve(__dirname, './bin/main.js'), (err, stats) => {
+            expect(err).toBe(null);
+            expect(stats.isFile()).toBe(true);
+            done();
+        });
+    });
+
+    it('should load a development config when --mode=production is passed', done => {
+        const { stderr, stdout } = run(__dirname, ['--prod', 'production']);
```
Comment (anshumanv) [inline test/mode/prod/prod.test.js]: Did you mean?

```suggestion
        const { stderr, stdout } = run(__dirname, ['--mode', 'production']);
```

File: test/mode/prod/prod.test.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        const { stderr, stdout } = run(__dirname, ['--mode', 'production', '--dev']);
+        expect(stderr).toContain('"mode" will be used');
+        expect(stdout).toBeTruthy();
+
+        stat(resolve(__dirname, './bin/main.js'), (err, stats) => {
+            expect(err).toBe(null);
+            expect(stats.isFile()).toBe(true);
+            done();
+        });
+    });
```
Comment (anshumanv) [inline test/mode/prod/prod.test.js]: You can add another case of using both --prod and --dev which should use prod config.

**Comment by ematipico** [comment]
@anshumanv thank you for your inputs! PR updated :)

**Pull request merged**